### ISSUE TITLE
[Bug]: Images don't load with new CDN URL in theme dev environment (shopify theme dev)

### ISSUE THREAD

**Issue by hedgehog90** [Shopify/cli#1859]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

New CDN URLs are being implemented for some Shopify stores it seems.

All images and asset URLs generated by the appropriate liquid filter used to have a prefix like:
`//cdn.shopify.com/s/files/.../products/`
But now, for some stores, they're like this:
`/cdn/shop/products/`

This brings with it additional problems which are already affecting many stores I've checked. I've detailed them in a separate issue in the Shopify Liquid repo, which gets little to no attention it seems:
https://github.com/Shopify/liquid/issues/1708
If someone could get the attention of someone who can help, please link them to this issue.

However, _this_ issue is specifically regarding the image assets loaded in shopify-cli dev.
A new CDN image URL looks like this:
`/cdn/shop/files/the-name-of-your-image.jpg`
Which the browser resolves to:
`http://127.0.0.1:9292/cdn/shop/files/the-name-of-your-image.jpg`
It would appear localhost will serve a 200 response but invalid image data so no image loads.
I've confirmed that some other theme devs have started experiencing this within the last 1-2 days.
However other assets like JavaScript and stylesheets and the like are served correctly through the new CDN URLs.
The only way to fix it currently is to prepend every `image_url` output with `shop.url`

### Actual behavior

Images don't load.

### Verbose output

```shell
-
```

### Reproduction steps

You need to have a Shopify store that uses the new CDN URLs. It seems most stores are still using the old CDN URLs as mentioned earlier. 

### Operating System

Windows 11

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.45.1

**Comment by amcaplan** [comment]
cc @Shopify/theme-developer-tools

**Comment by mgmanzella** [comment]
@Shopify/theme-code-tools

**Comment by Arkham** [comment]
hey @hedgehog90 thanks for reporting this, we have a team working on this. They've identified the root cause of the problem and are working on a fix.

**Comment by hedgehog90** [comment]
Is this still not fixed?

**Comment by jay-aptimized** [comment]
We've encountered a significant increase in 404 errors correlating directly with the appearance of /file/ in the URL paths across our Shopify site. Initially, our logs indicated a minor number of 404s, but following recent changes, these have escalated dramatically, specifically after /file/ was introduced into the URL structure. While Shopify CDN URLs containing /file/ are resolving as expected, any attempt to access URLs with /file/ on our actual site results in a 404 error.

This issue suggests there might be a routing or path resolution problem within the Shopify environment when /file/ is included in the URL path. It's unclear whether this is due to a recent update or a change in the way Shopify handles static assets or URL routing. We have verified our theme and asset configurations to ensure no unintentional changes have been made that could contribute to this issue.

Any insights or suggestions on how to resolve these unexpected 404 errors would be greatly appreciated. Is there a known issue with the handling of /file/ in URLs, or could there be a specific configuration required to support this path structure without resulting in 404s?
